sale_blanket_order/wizard/create_sale_orders.py

```diff
@@ -0,0 +1,205 @@
+                    line.order_id.name)
+
+            if float_is_zero(
+                    line.remaining_uom_qty, precision_digits=precision):
+                raise UserError(
+                    _('The sale has already been completed.'))
+
+            line_company_id = line.company_id and line.company_id.id or False
+            if company_id is not False \
+                    and line_company_id != company_id:
+                raise UserError(
+                    _('You have to select lines '
+                      'from the same company.'))
+            else:
+                company_id = line_company_id
+
+    @api.model
+    def _default_lines(self):
+        blanket_order_line_obj = self.env['sale.blanket.order.line']
+        blanket_order_line_ids = self.env.context.get('active_ids', False)
+        active_model = self.env.context.get('active_model', False)
+
+        if active_model == 'sale.blanket.order':
+            bo_lines = self._default_order().line_ids
+        else:
+            bo_lines = blanket_order_line_obj.browse(blanket_order_line_ids)
+
+        self._check_valid_blanket_order_line(bo_lines)
+
+        lines = [(0, 0, {
+            'blanket_line_id': l.id,
+            'product_id': l.product_id.id,
+            'date_schedule': l.date_schedule,
+            'remaining_uom_qty': l.remaining_uom_qty,
+            'price_unit': l.price_unit,
+            'product_uom': l.product_uom,
+            'qty': l.remaining_uom_qty,
+            'partner_id': l.partner_id,
+        }) for l in bo_lines]
+        return lines
+
+    blanket_order_id = fields.Many2one(
+        'sale.blanket.order', readonly=True)
+    sale_order_id = fields.Many2one(
+        'sale.order',
+        string='Purchase Order',
+        domain=[('state', '=', 'draft')])
+    line_ids = fields.One2many(
+        'sale.blanket.order.wizard.line', 'wizard_id',
+        string='Lines', default=_default_lines)
+
+    @api.multi
+    def create_sale_order(self):
+
+        order_lines_by_customer = defaultdict(list)
+        currency_id = 0
+        pricelist_id = 0
+        user_id = 0
+        payment_term_id = 0
+        for line in self.line_ids:
+            if line.qty == 0.0:
+                continue
+
+            if line.qty > line.remaining_uom_qty:
+                raise UserError(
+                    _('You can\'t order more than the remaining quantities'))
+
+            vals = {'product_id': line.product_id.id,
+                    'name': line.product_id.name,
+                    'product_uom': line.product_uom.id,
+                    'sequence': line.blanket_line_id.sequence,
+                    'price_unit': line.blanket_line_id.price_unit,
+                    'blanket_order_line': line.blanket_line_id.id,
+                    'product_uom_qty': line.qty,
+                    'tax_id': [(6, 0, line.taxes_id.ids)]}
+            order_lines_by_customer[line.partner_id.id].append((0, 0, vals))
+
+            if currency_id == 0:
```

Comment:
Why all those `if/elif`? `currency_id`, `pricelist_id`, `user_id`, `payment_term_id` are all fetched (in the end) by the order of the lines. Why not fetch them from the order since the lines here will belong to the same BO?